Lib/test/test__xxsubinterpreters.py

```diff
@@ -1337,6 +1337,97 @@ def test_run_string_arg_resolved(self):
         self.assertEqual(obj, b'spam')
         self.assertEqual(out.strip(), 'send')
 
+    def test_channel_send_wait_same_interpreter(self):
+        cid = interpreters.channel_create()
+
+        received = interpreters.channel_send_wait(cid, b"send", timeout=0)
+        self.assertFalse(received)
+
+        obj = interpreters.channel_recv(cid)
+        self.assertEqual(obj, b"send")
+
+    def test_channel_send_wait_different_interpreters(self):
+        cid = interpreters.channel_create()
+        interp = interpreters.create()
+        _run_output(interp, dedent(f"""
+            import _xxsubinterpreters as _interpreters
+            import time
+            import math
+
+            start = time.time()
+            rc = _interpreters.channel_send_wait({cid}, b"send", timeout=1)
+            end = time.time()
+
+            assert not rc
+            assert math.floor(end-start) == 1
+            """))
+
+        obj = interpreters.channel_recv(cid)
+        self.assertEqual(obj, b"send")
+
+    def test_channel_send_wait_different_threads_and_interpreters(self):
+        cid = interpreters.channel_create()
+        interp = interpreters.create()
+
+        thread_exc = None
+        def run():
+            try:
+                out = _run_output(interp, dedent(f"""
+                    import _xxsubinterpreters as _interpreters
+                    import time
+
+                    rc = _interpreters.channel_send_wait({cid}, b"send")
+                    assert rc
+                    """))
+            except Exception as e:
+                nonlocal thread_exc
+                thread_exc = e
+        t = threading.Thread(target=run)
+        t.start()
+        time.sleep(0.5)
+
+        obj = interpreters.channel_recv(cid)
+        self.assertEqual(obj, b"send")
+        t.join()
+        assert thread_exc is None, f"{thread_exc}"
+
+    def test_channel_send_wait_no_timeout(self):
+        cid = interpreters.channel_create()
+        interp = interpreters.create()
+
+        thread_exc = None
+        def run():
+            try:
+                out = _run_output(interp, dedent(f"""
+                    import _xxsubinterpreters as _interpreters
+                    import time
+
+                    rc = _interpreters.channel_send_wait({cid}, b"send", timeout=10)
```

Comment:
The test function says "no_timeout".  Did you mean "with_timeout"?  "timeout_not_hit"?